fix(insights): domain selector not working in eap

DominikB2014 opened the pull request:
Fixes several issues with the domain selector in eap

In eap, the `span.domain` field comes back as a string, whereas in the metrics it would come back as an array because the backend would split by `','` in the database. The eap rpc does not support splitting by `,`, therefore for now we have to do this in the frontend. To make the span.domain filter work, we wrap the domain value with wildcards.

For example:
1. span.domain for the db module comes back as `',table1,table2,'`
2. The frontend splits this result to produce `[table1, table2]` to populate the domain selector
3. when you select `table1` in the selector, the query gets populated with `span.domain:*table1*`

This fixes another issue where comma's would appear in the domain selector
Before
![image](https://github.com/user-attachments/assets/b9269dd8-2fb3-4220-bb2c-68424ee5c1d0)

After
![image](https://github.com/user-attachments/assets/60da37d9-63be-4ff0-ada1-587591c6813c)

DominikB2014 committed:
fix domain selector in eap

DominikB2014 committed:
fix test

DominikB2014 committed:
fix

DominikB2014 commented on static/app/views/insights/common/views/spans/selectors/domainSelector.tsx:
```diff
@@ -86,25 +88,41 @@ export function DomainSelector({
-  const incomingDomains = [
-    ...uniq(domainData?.flatMap(row => row[SpanMetricsField.SPAN_DOMAIN])),
+  const incomingDomains: Array<{label: string; value: string}> = [
+    ...uniq(
```
oops, i think this uniq won't work because we are now using an array of objects

DominikB2014 committed:
fix

gggritso commented on static/app/views/insights/common/views/spans/selectors/domainSelector.tsx:
```diff
@@ -87,24 +89,39 @@ export function DomainSelector({
 
+  const domainList = incomingDomains.map(domain => ({
+    label: domain,
+    value: useEap ? `*${domain}*` : domain,
```
I think this should be:
```suggestion
    value: useEap ? `*,${domain},*` : domain,
```

🤔 If there's a table called `app_users` and another table called `app_usersaccounts`, without commas, selecting `app_users` in the dropdown would search for `*app_users*` and return data from both tables. We need to search for `*,app_users,*` to make sure the filtering works correctly.

DominikB2014 replied:
Hmm, I don't think this will work for resource module, where span.domain is a single value containing no commas. Will confirm tmrw!

But regardless, I think I can update to work like this when there are commas

gggritso commented on static/app/views/insights/common/views/spans/selectors/domainSelector.tsx:
```diff
@@ -87,24 +89,39 @@ export function DomainSelector({
   if (value) {
-    incomingDomains.push(value);
+    const scrubbedValue = useEap
+      ? value.replace(/^\*(.*)\*$/, '$1') // removes wildcards, only if they are at the beginning and end of the string
```
Sorry, I'm not following, why is this needed?

DominikB2014 replied:
In the domain selector we add the current selected domain to the domain list,
`value` comes from the query params ` decodeScalar(location.query?.['span.domain'])`
i.e 
```
if (value {
domainList.push({
      label: scrubbedValue,
      value,
    });
}
```

In this case, when you select a domain, for example `table1`, the query param becomes `*table1*`.

If we added that back to the `domainList` directly, we would have an entirely new entry for `*table1*`. Therefore we need to remove the wildcards to prevent adding the wildcard version to the list.

gggritso reviewed (commented):
One important thing about commas that I _think_ is right, but overall makes sense!

Also, do the tables in the list need to be sorted?

DominikB2014 committed:
update

DominikB2014 commented:
@gggritso I addressed your comments, for db module we will add commas to surrounding the query, but we won't for the resource module. And also, the list already sorted!
<img width="425" alt="image" src="https://github.com/user-attachments/assets/a7f5cf0b-377b-4731-9a50-2a3a341c4c25" />

gggritso reviewed (approved):
👍🏻
